iothub/device/src/Exceptions/IotHubClientException.cs

```diff
@@ -37,65 +37,20 @@ protected internal IotHubClientException() : base()
         {
         }
 
-        /// <summary>
-        /// Creates an instance of this class.
-        /// </summary>
-        /// <param name="message">The message that describes the error.</param>
-        /// <param name="innerException">The exception that is the cause of the current exception.</param>
-        protected internal IotHubClientException(string message, Exception innerException = null)
```

Comment:
I see, because you removed overloads.

Well, I didn't love the many ctor overloads either. I'd just rather we use settable properties so the instantiator can set what they want, and make the required values be ctor params (probably message and inner exception).

Reply:
I don't mind that approach since the multiple optional params on the constructor didn't feel very user friendly to me. Having settable internal properties seems like a better option.